### PR TITLE
Add new image for user_saml_shibboleth-php8.0

### PR DESCRIPTION
**Pull request by blizzz**
🤞‍

* [x] fix       cURL error 7: Failed to connect to ::1: Cannot assign requested address (see http://curl.haxx.se/libcurl/c/libcurl-errors.html) (GuzzleHttp\Exception\ConnectException) (cf. https://drone.nextcloud.com/nextcloud/user_saml/465/1/2)
* [x] final test and stash

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

File: user_saml_shibboleth-php7.3/start.sh [hunk 1]
```diff
@@ -28,5 +28,10 @@ do
    fi
    sleep 1
 done
+if [ ${IS_LDAP} -ne 0 ]; then
+	echo "LDAP is not ready"
+	cat /var/log/dirsrv/slapd-dir/errors
+	exit 1
+fi
 
 run-jetty.sh
```
Comment (artonge) [inline user_saml_shibboleth-php7.3/start.sh:37]:
```suggestion
run-jetty.sh &
```

File: user_saml_shibboleth-php7.3/Dockerfile [hunk 2]
```diff
@@ -15,8 +15,19 @@ RUN sed -i 's/updateSelinuxPolicy($inf);//g' /usr/lib64/dirsrv/perl/*
...
 ADD ldap/AdminServer.pm /usr/lib64/dirsrv/perl/AdminServer.pm
 RUN setup-ds-admin.pl --silent --file /ds-setup.inf
-RUN /usr/sbin/ns-slapd -D /etc/dirsrv/slapd-dir && sleep 20 && ldapmodify -H ldap:/// -f nextcloud.ldif -x -D "cn=Directory Manager" -w password && ldapadd -H ldap:/// -f users.ldif -x -D "cn=Directory Manager" -w password
+RUN /usr/sbin/ns-slapd -D /etc/dirsrv/slapd-dir \
+	# ensure to have enough time for startup
+	&& sleep 40 \
+	&& ldapmodify -H ldap:/// -f nextcloud.ldif -x -D "cn=Directory Manager" -w password \
+	&& ldapadd -H ldap:/// -f users.ldif -x -D "cn=Directory Manager" -w password \
+	&& sleep 10 \
+	&& kill -INT "$(ps -C ns-slapd -o pid h)" \
```
Comment (artonge) [inline user_saml_shibboleth-php7.3/Dockerfile:24]: So, it looks like slapd is creating a file here with the process number: `/var/lock/dirsrv/slapd-dir/server`

Killing it with `INT` do not allow it to delete this file.
Killing it with `TERM` does.

```suggestion
 	&& kill -TERM "$(ps -C ns-slapd -o pid h)" \
```

Only tested it once, and the lock file is not here anymore so CI should be happier, crossing finger hard that it is 'simply' it :)

File: user_saml_shibboleth-php8.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ADD ldap/DSCreate.pm /usr/lib64/dirsrv/perl/DSCreate.pm
+ADD ldap/AdminServer.pm /usr/lib64/dirsrv/perl/AdminServer.pm
+RUN setup-ds-admin.pl --silent --file /ds-setup.inf
+RUN /usr/sbin/ns-slapd -D /etc/dirsrv/slapd-dir \
+    # ensure to have enough time for startup
+    && sleep 40 \
+    && ldapmodify -H ldap:/// -f nextcloud.ldif -x -D "cn=Directory Manager" -w password \
+    && ldapadd -H ldap:/// -f users.ldif -x -D "cn=Directory Manager" -w password \
+    && sleep 10 \
+    && kill -INT "$(ps -C ns-slapd -o pid h)" \
```
Comment (artonge) [inline user_saml_shibboleth-php8.0/Dockerfile:25]:
```suggestion
    && kill -TERM "$(ps -C ns-slapd -o pid h)" \
```

**Review by artonge — CHANGES_REQUESTED** [review]

**Review by artonge — APPROVED** [review]

**Comment by blizzz** [comment]
@artonge I think that was it. So far i have not seen a single failing test. Thank you so much!!

**Pull request merged**